server/src/main/java/org/opensearch/extensions/ExtensionsOrchestrator.java

```diff
@@ -333,6 +360,52 @@ TransportResponse handleRegisterTransportActionsRequest(RegisterTransportActions
         return new ExtensionBooleanResponse(true);
     }
 
+    /**
+     * Handles a {@link EnvironmentSettingsRequest}.
+     *
+     * @param environmentSettingsRequest  The request to handle.
+     * @return  A {@link EnvironmentSettingsResponse}
+     * @throws Exception if the request is not handled properly.
+     */
+    TransportResponse handleEnvironmentSettingsRequest(EnvironmentSettingsRequest environmentSettingsRequest) throws Exception {
+        return new EnvironmentSettingsResponse(this.initialEnvironmentSettings, environmentSettingsRequest.getComponentSettings());
+    }
+
+    /**
+     * Handles a {@link AddSettingsUpdateConsumerRequest}.
+     *
+     * @param addSettingsUpdateConsumerRequest  The request to handle.
+     * @return  A {@link ExtensionBooleanResponse} indicating success.
+     * @throws Exception if the request is not handled properly.
+     */
+    TransportResponse handleAddSettingsUpdateConsumerRequest(AddSettingsUpdateConsumerRequest addSettingsUpdateConsumerRequest)
+        throws Exception {
+
+        List<WriteableSetting> extensionComponentSettings = addSettingsUpdateConsumerRequest.getComponentSettings();
+        DiscoveryExtension extensionNode = addSettingsUpdateConsumerRequest.getExtensionNode();
+
+        for (WriteableSetting extensionComponentSetting : extensionComponentSettings) {
+
+            // Extract setting and type from writeable setting
+            Setting<?> setting = extensionComponentSetting.getSetting();
+            WriteableSetting.WriteableSettingGenericType settingType = extensionComponentSetting.getType();
+
+            // Register setting update consumer with callback method to extension
+            this.clusterService.getClusterSettings().addSettingsUpdateConsumer(setting, (data) -> {
+                logger.info("Sending extension request type: " + REQUEST_EXTENSION_UPDATE_SETTINGS);
+                UpdateSettingsResponseHandler updateSettingsResponseHandler = new UpdateSettingsResponseHandler();
+                transportService.sendRequest(
+                    extensionNode,
+                    REQUEST_EXTENSION_UPDATE_SETTINGS,
+                    new UpdateSettingsRequest(settingType, setting, data),
+                    updateSettingsResponseHandler
+                );
+            });
+        }
```

Comment:
Let's create separate class for the handlers

Reply:
Sure, I just added separate request handler classes for `AddSettingsUpdateConsumerRequest` and `EnvironmentSettingsRequest`